### PR TITLE
Fixes #101 Sorting and Pagination with tablesorter

### LINKED ISSUE #101: As an advisor, I need to sort lists of students...
As an advisor, I need to sort lists of students so that I can order the students in whatever way works best for me, particularly so that I can find students who are most at risk very easily.

Requirements, ordered from highest to lowest priority:

1. Sort "My Students" list by status (more details on this at bottom)
2. Sort "My Students" list by last name
3. Sort "My Students" list by first name

User should be able to click column header to sort that column A-Z, and click again to sort Z-A. Note that for status sort, this would mean default seeing students with "lowest" status first.

Not sure if helpful but I've used [DataTables](https://datatables.net/) in the past to create dynamically sortable tables.

Details about sorting by status:
- Currently, I believe there is no aggregation of student's statuses in courses, only individual statuses per student-course combination. Some sort of aggregation will need to happen in order to sort by status
- I would like to sort by the AVERAGE status--add up all the statuses they have and divide by the number of statuses. The reason for this is that it will sort the students who are the most "at risk" to the top and will not matter how many classes a student is taking. Would like students with no courses or no statuses to bottom.

### PR DESCRIPTION
**Pull request by jonespm**

### CODE REVIEW

File: seumich/templates/seumich/student_list_partial.html [hunk 1]
```diff
@@ -17,14 +19,54 @@ <h2 class="list-header">Students</h2>
...
+                                class="glyphicon glyphicon-backward"></span></button>
+                    </div>
+                    <span class="pagedisplay"></span>
+                    <div class="btn-group btn-group-sm" role="group">
+                        <button type="button" class="btn btn-default next"><span
+                                class="glyphicon glyphicon-forward"></span></button>
+                        <button type="button" class="btn btn-default last"><span
+                                class="glyphicon glyphicon-step-forward"></span></button>
+                    </div>
+                    <select class="form-control input-sm pagesize" title="Select page size">
```
Comment (zqian) [inline seumich/templates/seumich/student_list_partial.html]: like the idea for selecting page sizes!

File: seumich/static/seumich/sort_table.js [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+        // target the pager markup - see the HTML block below
+        container: $(".ts-pager"),
+        // target the pager page select dropdown - choose a page
+        cssGoto  : ".pagenum",
+        // remove rows from the table to speed up the sort of large tables.
+        // setting this to false, only hides the non-visible rows; needed if you plan to add/remove rows with the pager enabled.
+        removeRows: false,
+        // output string - default is '{page}/{totalPages}';
+        // possible variables: {page}, {totalPages}, {filteredPages}, {startRow}, {endRow}, {filteredRows} and {totalRows}
+        output: "{startRow} - {endRow} / {filteredRows} ({totalRows})"
```
Comment (zqian) [inline seumich/static/seumich/sort_table.js]: need to make this output line more readable?

As included in the Slack screenshot, this line is 1-7/7(7), which might be hard for user to understand.

File: seumich/templates/seumich/student_list_partial.html [hunk 3]
```diff
@@ -17,14 +19,54 @@ <h2 class="list-header">Students</h2>
...
+                        <option selected="selected" value="10">10</option>
+                        <option value="20">20</option>
+                        <option value="30">30</option>
+                        <option value="all">All Rows</option>
+                    </select>
+                    <select class="form-control input-sm pagenum" title="Select page number"></select>
```
Comment (zqian) [inline seumich/templates/seumich/student_list_partial.html]: Need label again for this dropdown. Also, I am not how useful it is for user to select page number directly from the drop down list. 

They've already go the arrows to the left for page navigation.

File: seumich/templates/seumich/student_list_partial.html (hunk 1 above)
Comment (jonespm) [reply seumich/templates/seumich/student_list_partial.html]: Yeah, same feature that was in Assignments! ;)

File: seumich/static/seumich/sort_table.js (hunk 2 above)
Comment (jonespm) [reply seumich/static/seumich/sort_table.js]: Yeah, we can improve this output line. I'll find a quick slot to do a demo today and improve this.

File: seumich/templates/seumich/student_list_partial.html (hunk 3 above)
Comment (jonespm) [reply seumich/templates/seumich/student_list_partial.html]: Yeah, probably not very useful unless there are a ton of pages. Even then, maybe not even . . . I don't know. I basically just copied the settings from the [Bootstrap 3 theme example](https://mottie.github.io/tablesorter/docs/example-option-theme-bootstrap-v3.html). But we can modify it to work better for us.

**Comment by jonespm** [comment]
FYI If you look at the example here:

https://mottie.github.io/tablesorter/docs/example-option-theme-bootstrap-v3.html

This is mostly what I based this on. There's a few things we can change up as mentioned by @zqian in the comments.

* We can change the styling or configuration of the pager on the bottom. It could use some more clarification or UX cleanup.
* We can identify some columns to have a "select" for the filter rather than a type entry. This is shown in the demo in the "Pick a gender" column with only two choices. "Maybe" we can do that for the status field. I'm not sure how to would work for the cohorts and might not make sense for the other fields since they would likely be unique. 
* This new table is going to be on pages other than the students page so maybe there are some tweaks on these other pages too.

I can schedule a time to talk to Maggie about this.

**Comment by jonespm** [comment]
At the call today, Maggie suggested we disable the filter for the "Status" since it's only seeming to pick up the first status. Also we could better clarify the pager. I forgot what else was suggested but I'll throw version 2 together.

**Comment by mfldavidson** [comment]
What we decided this morning was:
- filtering AND sorting on name and UMID columns
- sorting ONLY on status, but only once we can get it working with the algorithm Matt suggested in #101 
- filtering ONLY on cohort - we will release with the type-in-text-string type of filter and just see how people feel about it

**Comment by jonespm** [comment]
Ok, I implemented the suggestions here on the latest PR.

image

**Pull request merged**